astropy/units/quantity.py

```diff
@@ -1037,16 +1037,14 @@ def __getitem__(self, key):
         return out
 
     def __setitem__(self, i, value):
-        # update indices
-        if not self.isscalar:
+        # update indices if we're part of a table.
+        if not self.isscalar and 'info' in self.__dict__:
```

Comment:
Is `'info'` in `__dict__` some sort of duck-typing? If so could you maybe add this as comment that it's duck-typing?